stl/inc/expected

```diff
@@ -0,0 +1,1034 @@
+        : _Has_value(_Other._Has_value) {
+        if (_Has_value) {
+            _STD construct_at(_STD addressof(_Value), _STD forward<_Uty>(_Other._Value));
+        } else {
+            _STD construct_at(_STD addressof(_Unexpected), _STD forward<_UErr>(_Other._Unexpected));
+        }
+    }
+
+    // clang-format off
+    template <class _Uty = _Ty>
+        requires (!is_same_v<remove_cvref_t<_Uty>, in_place_t> && !is_same_v<remove_cvref_t<_Uty>, expected> //
+                 && !_Is_specialization_v<remove_cvref_t<_Uty>, unexpected> && is_constructible_v<_Ty, _Uty>)
+    constexpr explicit(!is_convertible_v<_Uty, _Ty>)
+        expected(_Uty&& _Other) noexcept(is_nothrow_constructible_v<_Ty, _Uty>) // strengthened
+        : _Value(_STD forward<_Uty>(_Other)), _Has_value(true) {}
+    // clang-format on
+
+    template <class _UErr>
+        requires is_constructible_v<_Err, const _UErr&>
+    constexpr expected(const unexpected<_UErr>& _Other) noexcept(
+        is_nothrow_constructible_v<_Err, const _UErr&>) // strengthened
+        : _Unexpected(_Other._Unexpected), _Has_value(false) {}
+
+    template <class _UErr>
+        requires is_constructible_v<_Err, _UErr>
+    constexpr expected(unexpected<_UErr>&& _Other) noexcept(is_nothrow_constructible_v<_Err, _UErr>) // strengthened
+        : _Unexpected(_STD forward<_UErr>(_Other._Unexpected)), _Has_value(false) {}
+
+    template <class... _Args>
+        requires is_constructible_v<_Ty, _Args...>
+    constexpr explicit expected(in_place_t, _Args&&... _Vals) noexcept(
+        is_nothrow_constructible_v<_Ty, _Args...>) // strengthened
+        : _Value(_STD forward<_Args>(_Vals)...), _Has_value(true) {}
+
+    // clang-format off
+    template <class _Uty, class... _Args>
+        requires is_constructible_v<_Ty, initializer_list<_Uty>&,_Args... >
+    constexpr explicit expected(in_place_t, initializer_list<_Uty> _Ilist, _Args&&... _Vals) noexcept(
+        is_nothrow_constructible_v<_Ty, initializer_list<_Uty>&, _Args...>) // strengthened
+        : _Value(_Ilist, _STD forward<_Args>(_Vals)...), _Has_value(true) {}
+    // clang-format on
+
+    template <class... _Args>
+        requires is_constructible_v<_Err, _Args...>
+    constexpr explicit expected(unexpect_t, _Args&&... _Vals) noexcept(
+        is_nothrow_constructible_v<_Err, _Args...>) // strengthened
+        : _Unexpected(_STD forward<_Args>(_Vals)...), _Has_value(false) {}
+
+    // clang-format off
+    template <class _Uty, class... _Args>
+        requires is_constructible_v<_Err, initializer_list<_Uty>&, _Args... >
+    constexpr explicit expected(unexpect_t, initializer_list<_Uty> _Ilist, _Args&&... _Vals) noexcept(
+        is_nothrow_constructible_v<_Err, initializer_list<_Uty>&, _Args...>) // strengthened
+        : _Unexpected(_Ilist, _STD forward<_Args>(_Vals)...), _Has_value(false) {}
+    // clang-format on
+
+    // [expected.object.dtor]
+    constexpr ~expected() noexcept {

```

Comment:
@CaseyCarter reminded us of the Clang bug where the order of constrained destructors is significant:
https://github.com/microsoft/STL/blob/314f65f831f827dbfb05ebcc0f08c496f5ddf64f/stl/inc/ranges#L280
The order you have here appears to be correct, but (1) I believe we should have the same comment here as this code pattern is equally affected and (2) we should ensure that we have `is_trivially_destructible` test coverage.

Reply:
There is already coverage for trivial destructors